Ssh/SshShell.php

```diff
@@ -161,25 +172,22 @@ public function connect($authenticate = true)
         }
     }
 
-    /**
-     * @throws \RuntimeException
-     */
-    protected function assertConnection()
+    protected function assertConnection($auth = true)
```

Comment:
Nitpicking: Double use of $auth (as property and as argument here), maybe rename this one?